src/ProgramTest.elm

```diff
@@ -1423,11 +1423,31 @@ queueSimulatedEffect effect programTest =
 
                         SimulatedEffect.PushUrl url ->
                             programTest
-                                |> routeChangeHelper ("simulating effect: SimulatedEffect.Navigation.pushUrl " ++ escapeString url) False url
+                                |> routeChangeHelper ("simulating effect: SimulatedEffect.Navigation.pushUrl " ++ escapeString url) 0 url
 
                         SimulatedEffect.ReplaceUrl url ->
                             programTest
-                                |> routeChangeHelper ("simulating effect: SimulatedEffect.Navigation.replaceUrl " ++ escapeString url) True url
+                                |> routeChangeHelper ("simulating effect: SimulatedEffect.Navigation.replaceUrl " ++ escapeString url) 1 url
+
+                        SimulatedEffect.Back n ->
+                            case state.navigation of
+                                Nothing ->
+                                    programTest
+
+                                Just { currentLocation, browserHistory } ->
+                                    if n <= 0 then
+                                        programTest
+
+                                    else
+                                        case List.head (List.drop (n - 1) browserHistory) of
+                                            Nothing ->
+                                                -- n is bigger than the history;
+                                                -- in this case, browsers ignore the request
```

Comment:
💭 Does it instead make sense to go as far back as we have entries? If we have a history of 3, and we try to go back 5, then it would leave us at the first entry, with a new history size of zero.

It's like mashing your back button -- eventually you get to the end and clicking does nothing.

Reply:
We would imagine that to be the case as well, but when we looked up the spec for `back`, it looks like it just ignores the spec if given an n > history length. We also confirmed that browsers ignore this request, so it seems that the behavior here is correct 🤷‍♀